src/PrintReceipt.ts

```diff
@@ -1,12 +1,98 @@
 import {loadAllItems, loadPromotions} from './Dependencies'
 
-export function printReceipt(tags: string[]): string {
-  return `***<store earning no money>Receipt ***
-Name：Sprite，Quantity：5 bottles，Unit：3.00(yuan)，Subtotal：12.00(yuan)
-Name：Litchi，Quantity：2.5 pounds，Unit：15.00(yuan)，Subtotal：37.50(yuan)
-Name：Instant Noodles，Quantity：3 bags，Unit：4.50(yuan)，Subtotal：9.00(yuan)
+function loadProductMap() {
+  const products = loadAllItems()
+  return new Map(
+    products.map(product => {
+      return [product.barcode, product]
+    }))
+}
+
+function renderRowReceipt(product: { unit: string; price: number; name: string; barcode: string }, qty: number, subtotal: number) {
+  return `Name：${product.name}，Quantity：${qty} ${product.unit}s，Unit：${product.price.toFixed(2)}(yuan)，Subtotal：${subtotal.toFixed(2)}(yuan)`
+}
+
+function renderTitle(receipt: { total: number; discountTotal: number; receipt: string }) {
+  const title = '***<store earning no money>Receipt ***'
+  receipt.receipt += title
+}
+
+function calculateSubtotal(product: { unit: string; price: number; name: string; barcode: string }, qty: number, isDiscount: boolean) {
+  if (isDiscount) {
```

Comment:
The discount is fixed to "BUY_TWO_GET_ONE_FREE"
It's better to write a BUY_TWO_GET_ONE_FREE discount function explicitly